tasks.py

```diff
@@ -300,6 +361,170 @@ def post_upgrade(context):
     run_command(context, command)
 
 
+@task(
+    help={
+        "service": "Docker compose service name to run command in (default: nautobot).",
+        "command": "Command to run (default: bash).",
+        "file": "File to run command with (default: empty)",
+    },
+)
+def exec(context, service="nautobot", command="bash", file=""):
+    """Launch a command inside the running container (defaults to bash shell inside nautobot container)."""
+    command = [
+        "exec",
+        "--",
+        service,
+        command,
+        f"< '{file}'" if file else "",
+    ]
+    docker_compose(context, " ".join(command), pty=not bool(file))
+
+
+@task(
+    help={
+        "db-name": "Database name (default: Nautobot database)",
+        "input-file": "SQL file to execute and quit (default: empty, start interactive CLI)",
+        "output-file": "Ouput file, overwrite if exists (default: empty, output to stdout)",
+        "query": "SQL command to execute and quit (default: empty)",
+    }
+)
+def dbshell(context, db_name="", input_file="", output_file="", query=""):
+    """Start database CLI inside the running `db` container.
+
+    Doesn't use `nautobot-server dbshell`, using started `db` service container only.
+    """
+    if input_file and query:
+        raise ValueError("Cannot specify both, `input_file` and `query` arguments")
+    if output_file and not (input_file or query):
+        raise ValueError("`output_file` argument requires `input_file` or `query` argument")
+
+    env = {}
+    if query:
+        env["_SQL_QUERY"] = query
+
+    command = [
+        "exec",
+        "--env=_SQL_QUERY" if query else "",
+        "-- db sh -c '",
+    ]
+
+    if _is_compose_included(context, "mysql"):
+        command += [
+            "mysql",
+            "--user=$MYSQL_USER",
+            "--password=$MYSQL_PASSWORD",
+            f"--database={db_name or '$MYSQL_DATABASE'}",
+        ]
+    elif _is_compose_included(context, "postgres"):
+        command += [
+            "psql",
+            "--username=$POSTGRES_USER",
+            f"--dbname={db_name or '$POSTGRES_DB'}",
+        ]
+    else:
+        raise ValueError("Unsupported database backend.")
+
+    command += [
+        "'",
+        '<<<"$_SQL_QUERY"' if query else "",
+        f"< '{input_file}'" if input_file else "",
+        f"> '{output_file}'" if output_file else "",
+    ]
+
+    docker_compose(context, " ".join(command), env=env, pty=not (input_file or output_file or query))
+
+
+@task(
+    help={
+        "input-file": "SQL dump file to replace the existing database with. This can be generated using `invoke backup-db` (default: `dump.sql`).",
+    }
+)
+def import_db(context, input_file="dump.sql"):
+    """Stop Nautobot containers and replace the current database with the dump into the running `db` container."""
+    docker_compose(context, "stop -- nautobot worker beat")
```

Comment:
```suggestion
    docker_compose(context, "stop -- nautobot worker beat")
    docker_compose(context, "down --volumes")
```

Reply:
Remove all volumes prior to importing new db data. Ref: https://github.com/nautobot/cookiecutter-nautobot-app/issues/38

Reply:
Can be surprising for users as there could potentially be other volumes, that we don't want to remove when importing DB.

e.g.: https://github.com/nautobot/nautobot-plugin-chatops/blob/develop/development/mattermost/docker-compose.yml#L28

Reply:
There is no need to remove database volume, as here is `DROP` in `dump.sql` file.

Reply:
@snaselj is the `DROP DATABASE` only in `dump.sql` files generated by using the `invoke backup-db` task? Because my `.sql` dump was generated with pg_dump from within the postgres container itself and did not contain `DROP DATABASE`. 

So maybe we still want to include this explicit `down --volumes` step for cases where a user's `dump.sql` file doesn't have `DROP DATABASE` in it, like how mine didn't.